accord-core/src/main/java/accord/impl/InMemoryCommandStore.java

```diff
@@ -459,45 +408,63 @@ void update(Ranges add)
         }
     }
 
-    class CFKLoader implements CommandLoader<TxnId>
+    static class CFKEntry extends TxnId
     {
-        private Command loadForCFK(TxnId data)
+        final boolean uninitialised;
+        public CFKEntry(TxnId copy, boolean uninitialised)
+        {
+            super(copy);
+            this.uninitialised = uninitialised;
+        }
+    }
+
+    class CFKLoader implements CommandLoader<CFKEntry>
+    {
+        final RoutableKey key;
+        CFKLoader(RoutableKey key)
+        {
+            this.key = key;
+        }
+
+        private Command loadForCFK(CFKEntry entry)
         {
-            GlobalCommand globalCommand = ifPresent(data);
+            GlobalCommand globalCommand = ifPresent(entry);
             if (globalCommand != null)
                 return globalCommand.value();
-            throw new IllegalStateException("Could not find command for CFK for " + data);
+            if (entry.uninitialised)
+                return uninitialised(entry);
+            throw new IllegalStateException("Could not find command for CFK for " + entry);
         }
 
         @Override
-        public TxnId txnId(TxnId txnId)
+        public TxnId txnId(CFKEntry txnId)
         {
             return loadForCFK(txnId).txnId();
         }
 
         @Override
-        public Timestamp executeAt(TxnId txnId)
+        public Timestamp executeAt(CFKEntry txnId)
         {
             return loadForCFK(txnId).executeAt();
         }
 
         @Override
-        public SaveStatus saveStatus(TxnId txnId)
+        public SaveStatus saveStatus(CFKEntry txnId)
         {
             return loadForCFK(txnId).saveStatus();
         }
 
         @Override
-        public List<TxnId> depsIds(TxnId data)
+        public List<TxnId> depsIds(CFKEntry data)
         {
             PartialDeps deps = loadForCFK(data).partialDeps();
             return deps != null ? deps.txnIds() : Collections.emptyList();
         }
 
         @Override
-        public TxnId saveForCFK(Command command)
+        public CFKEntry saveForCFK(Command command)
         {
-            return command.txnId();
+            return new CFKEntry(command.txnId(), command.saveStatus() == SaveStatus.Uninitialised);
```

Comment:
for consistency reasons

```suggestion
            return new CFKEntry(command.txnId(), command.saveStatus().isUninitialised();
```

Reply:
Changed